### ISSUE TITLE
Accessibility: Sort menu cannot be dismissed using Escape when in a modal

### ISSUE THREAD

**Issue by Blackbaud-ScottFreeman** [blackbaud/skyux-lists#48]
### Expected behavior
When a sort menu is in a modal, it should be able to be closed/dismissed using the Escape key.

### Actual behavior
Using the Escape key causes the modal to close.

### Steps to reproduce
Try out this demo
https://stackblitz.com/edit/sort-modal-accessibility

### Severity
Low

### Impact
Low
